src/System.CommandLine/Invocation/CliAction.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.CommandLine.Invocation;
+
+/// <summary>
+/// Defines a behavior associated with a command line symbol.
+/// </summary>
+public abstract class CliAction
+{
+    private protected CliAction()
+    {
+    }
+
+    /// <summary>
+    /// Indicates that the action terminates a command line invocation, and later actions are skipped.
+    /// </summary>
+    public bool Terminating { get; protected set; } = true;
```

Comment:
Middleware allowed for conditionally short-circuiting which is why I made this mutable, though I didn't test that it will actually work and suspect it won't, so making it readonly for now, until or unless we clarify the scenario, is probably a good idea.

Reply:
Changing this to `protected init`.

Reply:
No. We can revisit this implementation but I'd prefer not to block this PR on it as there are larger moving parts.